packages/firestore/src/core/query.ts

```diff
@@ -211,53 +209,45 @@ export function isCollectionGroupQuery(query: Query): boolean {
 }
 
 /**
- * Returns the implicit order by constraint that is used to execute the Query,
- * which can be different from the order by constraints the user provided (e.g.
- * the SDK and backend always orders by `__name__`).
+ * Returns the order by constraint that is used to execute the Query. Note that the implicit order by
+ * constraint can be different from the order by constraints the user provided (e.g. the SDK and
+ * backend always orders by `__name__`).
  */
 export function queryOrderBy(query: Query): OrderBy[] {
   const queryImpl = debugCast(query, QueryImpl);
   if (queryImpl.memoizedOrderBy === null) {
     queryImpl.memoizedOrderBy = [];
+    const explicitFields = new Set<string>();
```

Comment:
Nit: I wonder if we should use a different name for this variable. Because after adding explicit fields to this set, we also start adding implicit order-by fields to it. Something like `fieldsProcessed` or `fieldsAddedSoFar` or `result` or... something better 😆

Reply:
fieldsVisited? fieldsIncluded? fieldsNormalized? fieldsAdded?

Reply:
fieldsNormalized SGTM